crates/rnote-compose/src/builders/penpathmodeledbuilder.rs

```diff
@@ -164,65 +164,65 @@ impl PenPathModeledBuilder {
-            * MODELER_PARAMS.sampling_min_output_rate)
-            .ceil() as i32;
-
-        if n_steps > MODELER_PARAMS.sampling_max_outputs_per_call {
-            // If the no of outputs the modeler would need to produce exceeds the configured maximum
-            // (because the time delta between the last elements is too large), it needs to be restarted.
-            tracing::debug!(
-                "PenpathModeledBuilder: updating modeler with element failed,
-n_steps exceeds configured max outputs per call."
-            );
-
-            self.restart(element, now);
-        }
-        self.last_element_time = now;
-
-        let modeler_input = ModelerInput::new(
+        let modeler_input = ModelerInput {
             event_type,
-            (element.pos[0] as f32, element.pos[1] as f32),
-            now.duration_since(self.start_time).as_secs_f64(),
-            element.pressure as f32,
-            0.0,
-            0.0,
-        );
+            pos: (element.pos[0], element.pos[1]),
+            time: now.duration_since(self.start_time).as_secs_f64(),
+            pressure: element.pressure,
+        };
 
         match self.stroke_modeler.update(modeler_input) {
             Ok(results) => self.buffer.extend(results.into_iter().map(|r| {
-                let pos = r.pos();
-                let pressure = r.pressure();
-                Element::new(na::vector![pos.0 as f64, pos.1 as f64], pressure as f64)
+                let pos = r.pos;
+                let pressure = r.pressure;
+                Element::new(na::vector![pos.0, pos.1], pressure)
             })),
-            Err(e) => tracing::error!("Updating stroke modeler with element failed, Err: {e:?}"),
+            Err(e) => {
+                match e {
+                    ModelerError::Element {
+                        src: ElementError::Duplicate,
+                    } => return, // we have a duplicate element so go back
+                    ModelerError::Element {
+                        src: ElementError::NegativeTimeDelta,
+                    } => return, //error on times
+                    ModelerError::Element {
+                        src: ElementError::TooFarApart,
+                    } => {
+                        self.last_element = element;
+                        tracing::debug!(
+                            "PenpathModeledBuilder: updating modeler with element failed,
+            n_steps exceeds configured max outputs per call."
+                        );
+                        self.restart(element, now);
+                    }
+                    ModelerError::Element {
+                        src: ElementError::Order { src: _ },
+                    } => {
+                        self.last_element = element;
+                        tracing::error!("Updating stroke modeler with element failed, Err: {e:?}")
+                    }
+                    _ => return,
```

Comment:
Other ModelerError's that might get added in the future should be logged as well, instead of silently returning